src/cdk-experimental/ui-patterns/behaviors/event-manager/event-manager.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/**
+ * An event that supports modifier keys.
+ */
+export interface EventWithModifiers extends Event {
+  ctrlKey: boolean;
+  shiftKey: boolean;
+  altKey: boolean;
+  metaKey: boolean;
+}
+
+/**
+ * Options that are applicable to all event handlers.
+ */
+export interface EventHandlerOptions {
+  stopPropagation: boolean;
+  preventDefault: boolean;
+}
+
+/**
+ * A config that specifies how to handle a particular event.
+ */
+export interface EventHandlerConfig<T extends Event> extends EventHandlerOptions {
+  handler: (event: T) => Promise<boolean | void>;
+}
+
+/**
+ * Bit flag representation of the possible modifier keys that can be present on an event.
+ */
+export enum ModifierKey {
```

Comment:
Add an implementation comment about why we need a bitwise representation of these modifiers?

Reply:
I believe the goal here was to distinguish between key combinations (`Ctrl | Shift`) and having multiple modifier keys that accomplish the same thing (`[Ctrl, Shift]`). 

@mmalerba Could you confirm or explain further?

Reply:
Yep, `Ctrl | Shift` represents an event with both modifiers held, `[Ctrl, Shift]` represents an event with either modifier held (but not both)